src/reducers/url-state.js

```diff
@@ -53,6 +53,20 @@ const dataSource: Reducer<DataSource> = (state = 'none', action) => {
   }
 };
 
+//TODO: This is not used yet, see issue #222
+//This is for tracking mouse position in timeline-axis
+const mouseTimePosition: Reducer<Millseconds | null> = (
```

Comment:
There has been a few small mistakes here, and that's why Flow AND the test are failing:
1. You added the new state `mouseTimePosition` to `profileView.viewOptions.mouseTimePosition` and this is perfect.
2. In the selector, you access it from the right location too, good!
3. _but_ you added the reducer to `url-state.js`, moreover you don't export it (see the object at the end of the file). So it's never exposed.

So the fix should be that you implement the reducer in `src/reducers/proifle-view`, AND do not forget to expose it using the `combineReducers` function at the end of the file. I believe this will fix both Flow and the tests. And probably make the code work at all ahah :-)

What `combineReducers` does is no rocket science, it makes it easier to manage a complex state with nested objects, keeping the immutable principle: 
1. Every action will run all reducers.
2. Then combineReducers will take all results, and (recursively) changes the nested object _only_ if one of its properties changes.

The immutable principle: an object reference changes only and only if one or more of its properties changes. This makes comparison very cheap as we just have to check the reference (this is the `===` comparison).